packages/manager/apps/cloud/client/app/deskaas/deskaas-details/deskaas-details.routing.js

```diff
@@ -0,0 +1,47 @@
+export default /* @ngInject */($stateProvider) => {
+  $stateProvider.state('deskaas.details', {
+    url: '/:serviceName?action&token',
+    params: {
+      action: null,
+      token: null,
+      followTask: null,
+    },
+    component: 'deskaasDetailsComponent',
+    resolve: {
+      action: /* @ngInject */ $transition$ => $transition$.params().action,
+      followTask: /* @ngInject */ $transition$ => $transition$.params().followTask,
+      serviceName: /* @ngInject */ $transition$ => $transition$.params().serviceName,
+      token: /* @ngInject */ $transition$ => $transition$.params().token,
+      goToChangePassword: /* @ngInject */ ($state, serviceName) => () =>
+        $state.go('deskaas.details.change-password', { serviceName }),
+      goToConsoleAccess: /* @ngInject */ ($state, serviceName) => () =>
+        $state.go('deskaas.details.get-console-access', { serviceName }),
+      removeConfirmationParams: /* @ngInject */ ($location) => () => {
+        if ($location.$$search.action) {
```

Comment:
Instead you can use `squash` in the router parameter definition

https://ui-router.github.io/ng1/docs/latest/interfaces/params.paramdeclaration.html#squash